src/Orleans.Runtime/Hosting/GenericHostExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using Microsoft.Extensions.Hosting;
+
+namespace Orleans.Hosting
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHostBuilder"/>.
+    /// </summary>
+    public static class GenericHostExtensions
+    {
+        /// <summary>
+        /// Configures the host builder to host an Orleans silo.
+        /// </summary>
+        /// <param name="hostBuilder">The host builder.</param>
+        /// <param name="configureDelegate">The delegate used to configure the silo.</param>
+        /// <returns>The host builder.</returns>
+        /// <remarks>
+        /// Calling this method multiple times on the same <see cref="IHostBuilder"/> instance will result in one silo being configured.
+        /// However, the effects of <paramref name="configureDelegate"/> will be applied once for each call.
+        /// </remarks>
+        public static IHostBuilder AddOrleans(this IHostBuilder hostBuilder, Action<ISiloBuilder> configureDelegate)
```

Comment:
There should be an overload where you get a reference to the IConfiguration object in the delegate.

Reply:
Not sure if that's possible under this sub-builder pattern. The configuration can be pulled from the container or from the `hostBuilderContext`, so it can be used in `ConfigureServices(...)` calls, but it doesn't exist until `Build` is called on the `IHostBuilder`.

If we had separate containers for host & silo then it would be possible, since we could presumably access the host config while constructing the silo builder.

Reply:
You can just call the delegate inside the ConfigureServices of the Host, then it will be available. I think it should be possible to make decisions based on the environment and the configuration, e.g. which storage you want to use.

Reply:
@SebastianStehle are we allowed to call ConfigureServices from within ConfigureServices? I imagine it would throw from [here](https://github.com/aspnet/Extensions/blob/0de62ae930da31048ba7e54c8cd0c6c0bcbd9095/src/Hosting/Hosting/src/HostBuilder.cs#L197) since the collection would change during enumeration.

It could be made to work, though, by changing HostBuilder.

Reply:
Don't call ConfigureServices from within ConfigureServices. Invoke the second ConfigureServices action inline instead.

Reply:
See this commit: https://github.com/dotnet/orleans/pull/5261/commits/e36231d52b088eaefeba3cf9aa8ca3e9a923dd61